crates/data_components/src/duckdb/write.rs

```diff
@@ -113,9 +120,7 @@ impl DataSink for DuckDBDataSink {
         data: SendableRecordBatchStream,
         _context: &Arc<TaskContext>,
     ) -> datafusion::common::Result<u64> {
-        let mut num_rows = 0;
-
-        let mut db_conn = self.duckdb.connect().await.map_err(to_datafusion_error)?;
+        let mut db_conn = self.duckdb.connect_sync().map_err(to_datafusion_error)?;
```

Comment:
Why using sync method is necessary here?

Reply:
It's a setup for my next PR